### PR TITLE
web: Fix issues with the iframe polyfill

### PR DESCRIPTION
**Pull request by Toad06**
This PR simplifies the iframe polyfill (I think) and also:
- merges the mutation observers used for object/embed tags and iframe/frame tags.
- prevents some pages from not being displayed at all.
- should prevent console spamming on error, at least on the web extension, eg. https://www.y8.com/games/garfield_coop_catch
- applies the RufflePlayer config on time in self-hosted builds. The web extension will still usually apply the config too late.
- no longer polyfills iframes whose documentElement has the data-ruffle-optout attribute, just like the web extension does when the "ignore compatibility warnings" option is off.

Demo that reproduces most of these items:
[demo.zip](https://github.com/ruffle-rs/ruffle/files/5639234/demo.zip)

### CODE REVIEW

File: web/packages/core/src/polyfills.ts [hunk 1]
```diff
@@ -47,191 +36,164 @@ function replaceFlashInstances(): void {
...
-            (mutation) => mutation.addedNodes.length > 0
-        );
-        if (nodesAdded) {
-            replaceFlashInstances();
-        }
-    });
+function polyfillFrames(): void {
+    // Create live collections to track embed tags.
+    iframes = iframes || document.getElementsByTagName("iframe");
+    frames = frames || document.getElementsByTagName("frame");
```
Comment (Herschel) [inline web/packages/core/src/polyfills.ts:67]: Can we use `??` here?

**Comment by Herschel** [comment]
Thanks for cleaning this up!

For the "prevents some pages from not being displayed at all" case, could we add a test for this?

**Comment by Toad06** [comment]
Thanks for your feedback.

I've made a simpler test that reproduces this issue:
[iframes_onload.zip](https://github.com/ruffle-rs/ruffle/files/5647350/iframes_onload.zip)

I'll commit it if you think it's good.

File: web/packages/core/src/polyfills.ts (hunk 1 above)
Comment (Toad06) [reply web/packages/core/src/polyfills.ts:67]: I think `||` and `??` should be equivalent here. If it's better using `??`, let's do it then.
It's a change that could also be made to the object / embed tags, I think.

**Comment by Herschel** [comment]
The test looks good to me, thanks!

**Pull request merged**